src/platform/PICO/bus_spi_pico.c

```diff
@@ -321,39 +328,52 @@ void spiInternalResetStream(dmaChannelDescriptor_t *descriptor)
 bool spiInternalReadWriteBufPolled(SPI_TypeDef *instance, const uint8_t *txData, uint8_t *rxData, int len)
 {
     // TODO optimise with 16-bit transfers as per stm bus_spi_ll code
-    int bytesProcessed = spi_write_read_blocking(SPI_INST(instance), txData, rxData, len);
+    int bytesProcessed = 0;
+    if (txData && rxData) {
+        bytesProcessed = spi_write_read_blocking(SPI_INST(instance), txData, rxData, len);
+    } else if (txData) {
+        bytesProcessed = spi_write_blocking(SPI_INST(instance), txData, len);
+    } else if (rxData) {
+        // NB tx data: "Generally this can be 0, but some devices require a specific value here, e.g. SD cards expect 0xff" (pico-sdk spi.c).
+        uint8_t repeated_tx_data = 0xff; // cf. dummyTxByte in stm bus_spi_ll.c and for DMA here
+        bytesProcessed = spi_read_blocking(SPI_INST(instance), repeated_tx_data, rxData, len);
+    } else {
+        bprintf("\n*** unexpected spiInternalReadWriteBufPolled with no tx, rx");
+    }
+
     return bytesProcessed == len;
 }
 
-// Initialise DMA before first segment transfer
-void spiInternalInitStream(const extDevice_t *dev, bool preInit)
+void spiInternalInitStream(const extDevice_t *dev, volatile busSegment_t *segment)
 {
 #ifndef USE_DMA
     UNUSED(dev);
-    UNUSED(preInit);
+    UNUSED(segment);
 #else
-    UNUSED(preInit);
-
     busDevice_t *bus = dev->bus;
+    spi_inst_t *spi = SPI_INST(bus->busType_u.spi.instance);
 
-    volatile busSegment_t *segment = bus->curSegment;
+    // Prepare config, store in dmaInitTx/Rx, to be used in the following spiInternalStartDMA.
+    // To keep everything uniform (always both TX and RX channels active, callback always on RX completion), if there is
+    // no TX or RX buffer to read from / write to, treat as a single dummy byte with no increment.
+    // (cf. same idea on other platform implementations)
+    bool isTX = segment->u.buffers.txData;
+    bool isRX = segment->u.buffers.rxData;
```

Comment:
(bool conversion is not obvious here)
```suggestion
    bool isTX = segment->u.buffers.txData != NULL;
    bool isRX = segment->u.buffers.rxData != NULL;
```